src/web/crate_details.rs

```diff
@@ -233,10 +232,14 @@ impl CrateDetails {
     }
 }
 
-fn releases_for_crate(conn: &mut Client, crate_id: i32) -> Vec<Release> {
+fn releases_for_crate(
+    conn: &mut impl GenericClient,
+    crate_id: i32,
+) -> Result<Vec<Release>, postgres::Error> {
```

Comment:
These functions should all use `anyhow::Error` to get a backtrace attached as soon as possible.

Reply:
I didn't know about losing the backtrace here. 

I changed the return types, even though by doing that I had to switch to panicking in the data migration in `db::migrate`.